WireAPI/Sources/WireAPI/Authentication/AuthenticationManager.swift

```diff
@@ -29,56 +29,92 @@ protocol AuthenticationManagerProtocol {
 
 actor AuthenticationManager: AuthenticationManagerProtocol {
 
+    enum Failure: Error, Equatable {
+
+        case invalidCredentials
+
+    }
+
+    private enum CurrentToken {
+
+        case cached(AccessToken)
+        case renewing(Task<AccessToken, any Error>)
+
+    }
+
     private var currentToken: CurrentToken?
     private let clientID: String
-    private let cookieStorage: CookieStorage
+    private let cookieStorage: any CookieStorageProtocol
     private let networkService: NetworkService
 
     init(
         clientID: String,
-        cookieStorage: CookieStorage,
+        cookieStorage: any CookieStorageProtocol,
         networkService: NetworkService
     ) {
         self.clientID = clientID
         self.cookieStorage = cookieStorage
         self.networkService = networkService
     }
+    
+    /// Get a valid access token to make authenticated requests.
+    ///
+    /// If a valid token exists in the cache then it will be returned,
+    /// otherwise a new token will be retrieved from the backend.
+    ///
+    /// - Returns: A valid (non-expired) access token.
 
     func getValidAccessToken() async throws -> AccessToken {
         switch currentToken {
         case .renewing(let task):
+            // A new token will come soon, wait
             return try await task.value
 
         case .cached(let accessToken) where !accessToken.isExpiring:
```

Comment:
I don't think the `.isExpiring` case is tested right? I think it is also worth testing